### PR TITLE
Add the ability to ignore lines depending on comments

### PR DESCRIPTION
**Pull request by pasindud**
## Description
https://github.com/dart-lang/coverage - 0.14.0 added the ability to be able to ignore coverage depending on comments.

`// coverage:ignore-line` to ignore one line.
`// coverage:ignore-start` and `// coverage:ignore-end` to ignore range of lines inclusive.
`// coverage:ignore-file` to ignore the whole file.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [ ] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
This will be easier to land after https://github.com/flutter/flutter/pull/58658 updates the versions (we do it in a weird, lockstep kind of way)

**Comment by jonahwilliams** [comment]
@pasindud the version fixes have landed, you should be able to update and get this working now. Sorry for the delay!

**Comment by pasindud** [comment]
No problem, I have sync'd the PR

File: packages/flutter_tools/lib/src/test/coverage_collector.dart [hunk 1]
```diff
@@ -66,6 +66,11 @@ class CoverageCollector extends TestWatcher {
     assert(data != null);
 
     _logMessage('($observatoryUri): collected coverage data; merging...');
+    _addHitmap(await coverage.createHitmap(
+      data['coverage'] as List<Map<String, dynamic>>,
+      packagesPath: globalPackagesPath,
+      checkIgnoredLines: true,
+    ));
     _addHitmap(await coverage.createHitmap(data['coverage'] as List<Map<String, dynamic>>));
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/test/coverage_collector.dart:74]: This looks like you're adding the hitmap twice here and below

Comment (pasindud) [reply packages/flutter_tools/lib/src/test/coverage_collector.dart]: oh good catch, missed it during the re-sync

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Comment by jonahwilliams** [comment]
Thanks @pasindud , sorry this took a while to get though

**Comment by pasindud** [comment]
👍  Thanks for the help

**Pull request merged**